### ISSUE TITLE
[Question] feedback on feasibility of SGPR & PF using Edward

### ISSUE THREAD

**Issue by smcveigh-phunware** [blei-lab/edward#867]
I found Edward by viewing one of the Gaussian Process Summer School [videos](https://youtu.be/1zNNLHyeWok) demonstrating the COX process using a NBA player dataset [source](https://github.com/blei-lab/edward/blob/master/examples/cox_process.py).

Over the last couple of months, I've been learning about GPs, Sparse GPs, and bayesian optimization. A lot of it is still over my head for the moment. As such, I've been using both GPy & GPyOpt and GPflow & GPflowOpt for my project. The useful advantage of  these projects is that with trial and error I've found a kernel and mean function that works well with the my datasets.  The downsides are that GPflow (at least in my view) is to complicated an API to modify for doing anything other than GPs and GPy is not ideal for the future of my project.

My project is very similar to the COX example with the exception of a Gaussian prior, a Student-t likelihood (Gaussian acceptable for POC), and independent outputs on the same training data (each GP is an independent process). Ultimately, my goal is to find argmax_x P(X | Y_new) = argmax product( P_i(X | Y_new) for i in I) where I are the set of processes and Y_new is a new set of measurements.

That's why I was excited to see Edward. It seems like it provides a solid base wherein I could build on top a GP regression [1] that a particle filter [2] could interrogate, all inside of a TensorFlow framework.

[1] [Sparse Gaussian Processes Using Pseudo Inputs](https://papers.nips.cc/paper/2857-sparse-gaussian-processes-using-pseudo-inputs)
[2] [Kullback-Leibler divergence based improved particle filter](http://ieeexplore.ieee.org/document/6808793/)

I see from the Edward examples how to create a basic kernel and do variational inference with KLqp, which can keep me busy for the time being, but I'd appreciate any advice anyone here could give to on accomplishing the above goal described above.
